odoo/models.py

```diff
@@ -5851,6 +5851,33 @@ def filtered(self, func):
             self.mapped(name)
         return self.browse([rec.id for rec in self if func(rec)])
 
+    def _group_on(self, key):
+        """Eagerly groups the records of ``self`` by the ``key``, returning a
+        dict from the ``key``'s result to recordsets. All the resulting
+        recordsets are guaranteed to be part of the same prefetch-set.
+
+        Provides a convenience method to partition existing recordsets without
+        the overhead of a :meth:`~.read_group`, but performs no aggregation.
+
+        .. note:: unlike :func:`itertools.groupby`, does not care about input
+                  ordering, however the tradeoff is that it can not be lazy
+
+        :param key: either a callable from a :class:`Model` to a (hashable)
+                    value, or a field name. In the latter case, it is equivalent
+                    to ``itemgetter(key)`` (aka the named field's value)
+        :type key: callable | str
+        :rtype: dict
+        """
+        if isinstance(key, str):
+            key = itemgetter(key)
```

Comment:
Suggest to support grouping by multiple keys, either by provided a string of fields, separated by comma, or to input `key` as `list` or `tuple`

```suggestion
            key = itemgetter(*(k.strip() for k in key.split(",")))
```